packages/compiler/test/visibility.test.ts

```diff
@@ -964,6 +965,195 @@ describe("compiler: visibility core", () => {
       strictEqual(idRead.type, idReadCreate.type);
     });
   });
+
+  it("deeply renames types using the name template", async () => {
+    const { DataA, DataB } = (await runner.compile(`
+      enum Example {
+        A,
+        B,
+      }
+
+      model Data {
+        @visibility(Example.A)
+        data_a: Foo;
+
+        @visibility(Example.B)
+        data_b: Foo;
+      }
+        
+      model Foo {
+        @visibility(Example.B)
+        foo_b: string;
+        @visibility(Example.A)
+        foo_a: string;
+      }
+
+      @withVisibilityFilter(#{ any: #[Example.A] }, "{name}A")
+      @test model DataA {
+        ...Data
+      }
+
+      @withVisibilityFilter(#{ any: #[Example.B] }, "{name}B")
+      @test model DataB {
+        ...Data
+      }
+    `)) as { DataA: Model; DataB: Model };
+
+    ok(DataA);
+    ok(DataB);
+
+    ok(DataA.properties.has("data_a"));
```

Comment:
Does vitest not have a `expect().toHaveKey` or somethiung like that that gives more insight when it doesn't pass?